test/controllers/todos_controller_test.rb

```diff
@@ -989,6 +989,17 @@ def test_url_with_slash_in_query_string_are_parsed_correctly
     assert_select "a[href='#{url}']"
   end
 
+  def test_url_with_double_hyphen_is_parsed_correctly
```

Comment:
Should I move the other tests too? I did not know about `rendering_helper` when writing the test, by the way.